### ISSUE TITLE
Use of alt+enter to toggle fullscreen conflicts with default control scheme of Strife, Heretic, and Hexen

### ISSUE THREAD

**Issue by haleyjd** [chocolate-doom/chocolate-doom#988]
### Background

Version of Chocolate Doom: 3.0.0

Operating System and version: Any

Game: Heretic, Hexen Strife

Any loaded WADs and mods (please include full command line): None required

Alt+Enter is now allowed to toggle fullscreen status. However, in all three non-Doom games, alt is the default strafe key and enter is the default inventory use key. This means that if you attempt to use inventory while strafing, the game will abruptly change out of windowed mode without any intention on the user's part. This was always a major problem with playing these games in DosBox and other environments that support the same key combo without any means of disabling it.

Observed behavior: Windowed program changes to fullscreen.

Expected behavior: Using inventory item while strafing should be possible w/o causing a display mode set.

**Comment by jmtd** [comment]
What do you suggest instead?

I can't remember if we are consistent on all platforms but on Mac I think the command key is used either instead or as well as alt.

**Comment by Kroc** [comment]
Cmd+F is typical on the Mac and F11 is used in many instances on Windows.

**Comment by chungy** [comment]
F11 is a very bad option itself, it would override the gamma key.

Alt+Enter is a fairly de-facto standard in these matters, I believe it's pretty ok at least by default, since I suspect most players will be rebinding their keys anyway so it won't ever come up as an issue.

Perhaps it should be a function that can be rebound, but we presently don't have a way to bind actions to combinations of keys, and I feel like that would be a bit of a requirement here.

**Comment by jmtd** [comment]
To be honest it is something I regularly hit by accident if I haven't rebound so I appreciate the issue.  We aren't stuck with alt+enter for vanilla compat. Alt+f seems a fine choice

**Comment by fragglet** [comment]
See #696 for some past discussion. @linguica who originally lobbied for alt+enter over windows+enter.

As I see it, the options are either alt+enter, windows+enter, or alt-f.

Alt+enter thoughts:
* Already well-established since it was the key combo used under DOS, which arguably makes it more obvious
* Most people use WASD or some other dedicated strafe keys nowadays
* Same problem already existed under DOS

Windows+enter (or Cmd+enter on Mac):
* Solves the problem but is non-obvious

Alt+F (or Ctrl-Cmd+F on Mac?):
* Might be more appropriate nowadays since it's used in other apps anyway - web browsers?
* If people have custom key bindings it might conflict. For example suppose someone has F bound to "use item" since it's next to WASD, suddenly the same problem occurs.

I have a fourth suggestion: how about we keep alt+enter, but only make it work when the menu is open? It should be easy since already have the hooks used to release the mouse grab in windowed mode. That keeps the "expected" key binding, should avoid any gameplay conflicts, and is consistent with the windowed behavior.

**Comment by munday1** [comment]
F is also bound to "follow" in setup anyway...

**Comment by munday1** [comment]
I meant "follow" in automap.

**Comment by linguica** [comment]
I am still strongly in favor of retaining alt-enter - that key combination is as authentic to the time period as anything can be. My suggesting would be to leave it defaulted to alt+enter, but in the setup tool, under Keyboard configuration > Misc., add a new toggle button for "Use cmd/win-enter instead of alt-enter". As @fragglet says, this retains the "well-established" alt-enter as default, but in the setup utility in the obvious place, a user can switch it to cmd-enter which is otherwise "non-obvious."

**Comment by fragglet** [comment]
I agree that alt-enter is well-established and it would be nice to keep it if possible. But I'd prefer to avoid adding options to paper over the problem; this seems like an example of ["unbreak my application please"](http://ometer.com/free-software-ui.html).

Disabling alt+enter during gameplay but allowing it when the game is paused / menu is open seems reasonable. Does anyone have an objection?

**Comment by jmtd** [comment]
My only concern is it's undiscoverable.

**Comment by jmtd** [comment]
But the advantage probably outweighs that

**Comment by linguica** [comment]
The one main concern I can think of is that opening the menu will desync demo recording so a person can’t full screen / window during recording. But I don’t think that’s a particularly big loss.

**Comment by chungy** [comment]
> I have a fourth suggestion: how about we keep alt+enter, but only make it work when the menu is open? It should be easy since already have the hooks used to release the mouse grab in windowed mode. That keeps the "expected" key binding, should avoid any gameplay conflicts, and is consistent with the windowed behavior.

I like this suggestion. I never switch modes without pausing the game anyway.
